Libraries/src/Amazon.Lambda.KinesisAnalyticsEvents/KinesisAnalyticsFirehoseInputPreprocessingEvent.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Amazon.Lambda.KinesisAnalyticsEvents
+{
+    /// <summary>
+    /// This class represents the event from Kinesis Analytics application to preprocess Kinesis Firehose data.
+    /// </summary>
+    [DataContract]
+    public class KinesisAnalyticsFirehoseInputPreprocessingEvent
+    {
+        /// <summary>
+        /// Gets or sets the invocation identifier.
+        /// </summary>
+        /// <value>
+        /// The invocation identifier.
+        /// </value>
+        [DataMember(Name = "invocationId")]
+        public string InvocationId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the application arn.
+        /// </summary>
+        /// <value>
+        /// The application arn.
+        /// </value>
+        [DataMember(Name = "applicationArn")]
+        public string ApplicationArn { get; set; }
+
+        /// <summary>
+        /// Gets or sets the stream arn.
+        /// </summary>
+        /// <value>
+        /// The stream arn.
+        /// </value>
+        [DataMember(Name = "streamArn")]
+        public string StreamArn { get; set; }
+
+        /// <summary>
+        /// Gets or sets the records.
+        /// </summary>
+        /// <value>
+        /// The records.
+        /// </value>
+        [DataMember(Name = "records")]
+        public IList<FirehoseRecord> Records { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [DataContract]
+        public class FirehoseRecord
+        {
+
+            /// <summary>
+            /// Gets or sets the record identifier.
+            /// </summary>
+            /// <value>
+            /// The record identifier.
+            /// </value>
+            [DataMember(Name = "recordId")]
+            public string RecordId { get; set; }
+
+            /// <summary>
+            /// Gets or sets the record metadata.
+            /// </summary>
+            /// <value>
+            /// The record metadata.
+            /// </value>
+            [DataMember(Name = "kinesisFirehoseRecordMetadata")]
+            public KinesisFirehoseRecordMetadata RecordMetadata { get; set; }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            [DataContract]
+            public class KinesisFirehoseRecordMetadata
+            {
+                /// <summary>
+                /// Gets or sets the approximate arrival timestamp.
+                /// </summary>
+                /// <value>
+                /// The approximate arrival timestamp.
+                /// </value>
+                [DataMember(Name = "approximateArrivalTimestamp")]
+                public long ApproximateArrivalTimestamp { get; set; }
```

Comment:
Returning longs for dates in .NET makes it hard to use. Users have to understand that it is an Epoch and figure out how to convert from Epoch to DateTime.

We should do what we did for the KinesisFirehoseEvent and rename this property to ApproximateArrivalEpoch and then provide a second read only property called ApproximateArrivalTimestamp that converts the Epoch property to DateTime. Here is the code on how we did this before.

https://github.com/aws/aws-lambda-dotnet/blob/master/Libraries/src/Amazon.Lambda.KinesisFirehoseEvents/KinesisFirehoseEvent.cs#L50-L66

Reply:
Wouldn't be cleaner to have the property of the dto of type DateTimeOffset and use a JsonConverter responsible for converting to/from Unix epoch?

Reply:
Done.